src/andromede/simulation/optimization.py

```diff
@@ -494,33 +541,35 @@ def _create_objective(
     instantiated_expr = _instantiate_model_expression(
         objective_contribution, component.id, opt_context
     )
-    # We have already checked in the model creation that the objective contribution is neither indexed by time nor by scenario
-    linear_expr = component_context.linearize_expression(0, 0, instantiated_expr)
+    value_provider = _make_value_provider(opt_context, 0, 0, component)
```

Comment:
Lines 544 - 558 are very similar than in `_create_constraints` function, may be refactorized

Reply:
Yes it's really ugly :) needs more work

Reply:
I tried to clean up a little the implementation of problem building:
 - common operations such as expansion and linearization are provided by the context (which has the various "providers" needed)
 - I removed the need to use fake time and scenario index "0" for expressions that don't depend on time or scenario: we now use None to mean "time-independent" or "scenario-independent" and get an error if there is actually a time-dependent term in the expression